dpe/src/printer.rs

```diff
@@ -0,0 +1,33 @@
+// Licensed under the Apache-2.0 license
+
+use ufmt::uWrite;
+
+use platform::{Platform, PlatformError};
+
+use crate::dpe_instance::DpeTypes;
+
+pub struct Printer<'a, T: DpeTypes + 'a> {
+    pub platform: T::Platform<'a>,
+}
+
+impl<'a, T: DpeTypes> uWrite for Printer<'a, T> {
+    type Error = PlatformError;
+
+    fn write_str(&mut self, str: &str) -> Result<(), Self::Error> {
+        self.platform.write_str(str)?;
+        Ok(())
+    }
+}
+
+impl<'a, T: DpeTypes> Printer<'a, T> {
+    pub fn new(platform: T::Platform<'a>) -> Self {
```

Comment:
fyi: you might need to add this

```suggestion
    #[allow(dead_code)]
    pub fn new(platform: T::Platform<'a>) -> Self {
```

Reply:
Changed to _new instead.

Reply:
That can be a little awkward when we do use it though. Or do we always expect to use `plat_println`?

Reply:
I think we will always use plat_println! but when we do use plat_println! for the first time, _new will no longer be considered unused by the compiler and we can change it back to new

Reply:
I guess I was picturing it as more of a debugging capability than a regular addition. How often do we expect to commit code with `plat_println!`s? I'm good with the `_new` in that case then.

Reply:
I think you're right, we don't really expect to use it in production code. Changed to #[allow(dead_code)].

Reply:
Sorry to bring this back, but if we make the printer module public, because this is a public function it doesn't need #[allow(dead_code)]

Reply:
Done